File: Difficulty: Easy/Bus Ticket Change/bus-ticket-change.cpp
class Solution {
  public:
    bool canServe(vector<int> &arr) {
        int five_c = 0;
        int ten_c = 0;

        for (int i : arr) {
            if (i == 5) {
                five_c++;
            }
            else if (i == 10) {
                if (five_c == 0) return false;
                five_c--;
                ten_c++;
            }
            else { 
                if (ten_c >= 1 && five_c >= 1) {
                    ten_c--;
                    five_c--;
                }
                else if (five_c >= 3) {
                    five_c -= 3;
                }
                else {
                    return false;
                }
            }
        }
        return true;
    }
};

File: Difficulty: Easy/Bus Ticket Change/bus-ticket-change.java
class Solution {
    public boolean canServe(int[] arr) {
        // code here
        if(arr[0]>5) return false;
        
        int n = arr.length;
        int five_count = 1;
        int ten_count = 0;
        for(int i=1; i<n; i++){
            if(arr[i] == 5) {
                five_count += 1;
            }
            if(arr[i] == 10){
                five_count -= 1;
                ten_count += 1; 
            }
            if(arr[i] == 20){
                if(ten_count > 0) {
                    ten_count -= 1;
                }
                else{
                    five_count -= 2;
                }
                five_count -= 1;
            }
            if (five_count < 0 || ten_count<0){
                return false;
            }
        }
        return true;
    }
}